lighthouse-core/gather/gatherers/seo/font-size.js

```diff
@@ -0,0 +1,219 @@
+  });
+
+  return matchingRule;
+}
+
+/**
+ * @param {!Node} node
+ * @returns {!number}
+ */
+function getNodeTextLength(node) {
+  return !node.nodeValue ? 0 : node.nodeValue.trim().length;
+}
+
+/**
+ * @param {!Object} driver
+ * @param {!Node} node text node
+ * @returns {WebInspector.CSSStyleDeclaration}
+ */
+function getFontSizeSourceRule(driver, node) {
+  return driver.sendCommand('CSS.getMatchedStylesForNode', {nodeId: node.nodeId})
+    .then(matchedRules => getEffectiveRule(FONT_SIZE_PROPERTY_NAME, node, matchedRules));
+}
+
+/**
+ * @param {!Object} driver
+ * @param {!Node} node text node
+ * @returns {!{fontSize: number, textLength: number, node: Node}}
+ */
+function getFontSizeInformation(driver, node) {
+  return driver.sendCommand('CSS.getComputedStyleForNode', {nodeId: node.parentId})
+    .then(result => {
+      const {computedStyle} = result;
+      const fontSizeProperty = computedStyle.find(({name}) => name === FONT_SIZE_PROPERTY_NAME);
+
+      return {
+        fontSize: parseInt(fontSizeProperty.value, 10),
+        textLength: getNodeTextLength(node),
+        node: node.parentNode,
+      };
+    });
+}
+
+/**
+ * @param {Node} node
+ * @returns {boolean}
+ */
+function isNonEmptyTextNode(node) {
+  return node.nodeType === global.Node.TEXT_NODE &&
+    !TEXT_NODE_BLOCK_LIST.has(node.parentNode.nodeName) &&
+    getNodeTextLength(node) > 0;
+}
+
+class FontSize extends Gatherer {
+  /**
+   * @param {{driver: !Object}} options Run options
+   * @return {!Promise<{totalTextLength: number, failingTextLength: number, visitedTextLength: number, analyzedFailingTextLength: number, analyzedFailingNodesData: Array<{fontSize: number, textLength: number, node: Node, cssRule: WebInspector.CSSStyleDeclaration}>}>} font-size analysis
+   */
+  afterPass(options) {
+    const stylesheets = new Map();
+    const onStylesheetAdd = sheet => stylesheets.set(sheet.header.styleSheetId, sheet.header);
+    options.driver.on('CSS.styleSheetAdded', onStylesheetAdd);
+
+    const enableDOM = options.driver.sendCommand('DOM.enable');
+    const enableCSS = options.driver.sendCommand('CSS.enable');
+
+    let failingTextLength = 0;
+    let visitedTextLength = 0;
+    let totalTextLength = 0;
+
+    return Promise.all([enableDOM, enableCSS])
+      .then(() => getAllNodesFromBody(options.driver))
+      .then(nodes => {
+        const textNodes = nodes.filter(isNonEmptyTextNode);
+        totalTextLength = textNodes.reduce((sum, node) => sum += getNodeTextLength(node), 0);
+        const visitedNodes = textNodes
+          .sort((a, b) => getNodeTextLength(b) - getNodeTextLength(a))
+          .slice(0, MAX_NODES_VISITED);
+        visitedTextLength = visitedNodes.reduce((sum, node) => sum += getNodeTextLength(node), 0);
+
+        return visitedNodes;
+      })
+      .then(textNodes =>
+        Promise.all(textNodes.map(node => getFontSizeInformation(options.driver, node))))
```

Comment:
for these Promise.all's we'll probably want to make sure one missing node doesn't tank the whole gatherer

maybe a `.catch(err => { require('../lib/sentry.js').captureException(err); return null; })` on the `getFontSizeInformation` coupled with a `.filter(Boolean)` on `fontSizeInfo` in the .then?

failed on cnn for me
![image](https://user-images.githubusercontent.com/2301202/33563878-ed91a7e4-d8cd-11e7-9eaa-59a17007e63e.png)

Reply:
> I observed that the same thing happens for mutation events gatherer:

Yeah great point, we need to handle it there too :)

> BTW2 .filter(Boolean) is really fancy ✨ but IMHO .filter(x => !!x) is more readable and, as far as I can see, 20% faster (microbenchmarking FTW 😉).

well TIL! yeah sure pick your favorite `.filter` works for me :)
crazy that using built-in cast function is slower than creating a brand new lexical scoped function